test/compile_errors.zig

```diff
@@ -22,6 +22,46 @@ pub fn addCases(cases: *tests.CompileErrorContext) void {
         ".tmp_source.zig:3:28: error: @handle() in non-async function",
     );
 
+    cases.add(
+        "`_` is not a declarable symbol",
+        \\export fn f1() usize {
+        \\    var _: usize = 2;
+        \\    return _;
+        \\}
+    ,
+        ".tmp_source.zig:2:5: error: `_` is not a declarable symbol",
+        ".tmp_source.zig:3:12: error: use of undeclared identifier '_'",
```

Comment:
This is fine for now, but just a note that after #111 this second error would probably go away - in general we try to not emit errors that have a root cause of another error. So the 2nd error test is not necessary (but it's fine, we can remove it later).